### PR TITLE
Forms using form_for method

### PR DESCRIPTION
**Pull request by LoboArkano**
On this project, We showed all the knowledge gained through reading the material of The Odin Project section about Forms, Model and Validations.

**Project milestones**
- [x] Basic form using Rails helper to create users with username, password and email.
- [x] Basic form using Rails helper to update an existing user.
- [x] Show validations errors.

**General requirements:**

- [x] Add a descriptive README file to your repository
- [x] The README must include instructions on how to use the project on your machine
- [x] Use descriptive commit messages
- [x] Do not commit debugging instructions

**GitFlow:**

- [x] Create a development branch and set it as a default branch in your repository.
- [x] Create a feature-branch branch (give it a descriptive name)
- [x] Use Rubocop to validate your code.

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -0,0 +1,5 @@
+class User < ApplicationRecord
+  validates :username, presence: true, length: { maximum: 4 }
+  validates :email, presence: true
+  validates :password, presence: true
```
Comment (suretrust) [inline app/models/user.rb]: These two could be merged as `validates :email, :password, presence: true`

File: app/controllers/users_controller.rb [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+      @user.save
+      redirect_to new_user_path
+    else
+      render :new
+    end
+  end
+
+  def edit
+    @user = User.find(params[:id])
+    @user_temp = User.new
```
Comment (suretrust) [inline app/controllers/users_controller.rb]: You don't need this to edit your user. Just `@user = User.find(params[:id])` is cool.

File: app/controllers/users_controller.rb [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+  end
+
+  def update
+    @user = User.find(params[:id])
+    @user_temp = User.new(user_params)
```
Comment (suretrust) [inline app/controllers/users_controller.rb]: You don't need this to update your user. Just `@user = User.find(params[:id])` is cool.

File: app/views/users/edit.html.erb [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+  <%= f.text_field :username, value: "u3" %><br><br>
+  <%= f.label :password, "Password:"%>
+  <%= f.password_field :password%><br><br>
+  <%= f.label :email, "Email:"%>
+  <%= f.text_field :email, placeholder: "example@example.com"%><br><br>
+
+  <%= f.submit "Update"%>
+<%end%>
+
+<% if @user_temp.errors.any? %>
```
Comment (suretrust) [inline app/views/users/edit.html.erb]: `<% if @user_temp.errors.any? %>` means you are creating a new user entirely and checking if all fields are present. Use `<% if @user.errors.any? %>` instead, that way, you validate the exact user that you are editing.

Also, don't forget to replace `@user_temp` with `@user` in other places too.

File: app/controllers/users_controller.rb [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+
+    if @user_temp.valid?
```
Comment (suretrust) [inline app/controllers/users_controller.rb]: Use `if @user.valid?` instead.

File: app/controllers/users_controller.rb [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+      @user.update(user_params)
+      redirect_to new_user_path
```
Comment (suretrust) [inline app/controllers/users_controller.rb]: You should redirect to `edit_user_path`. That's the edit form, the `new_user_path` has a form that is empty and won't have the initial parameters of the user you are editing.

File: app/views/users/edit.html.erb [hunk 7]
```diff
@@ -0,0 +1,23 @@
+
+<%= form_for @user do |f|%>
+  <%= f.label :username, "Username:"%>
+  <%= f.text_field :username, value: "u3" %><br><br>
+  <%= f.label :password, "Password:"%>
+  <%= f.password_field :password%><br><br>
+  <%= f.label :email, "Email:"%>
+  <%= f.text_field :email, placeholder: "example@example.com"%><br><br>
```
Comment (suretrust) [inline app/views/users/edit.html.erb]: This should be `f.email_field`.

File: app/views/users/new.html.erb [hunk 8]
```diff
@@ -0,0 +1,26 @@
+
+<%= form_for @user do |f|%>
+  <%= f.label :username, "Username:"%>
+  <%= f.text_field :username, value: "u3" %><br><br>
+  <%= f.label :password, "Password:"%>
+  <%= f.password_field :password%><br><br>
+  <%= f.label :email, "Email:"%>
+  <%= f.text_field :email, placeholder: "example@example.com"%><br><br>
```
Comment (suretrust) [inline app/views/users/new.html.erb]: This should be `f.email_field`.

**Review by suretrust — CHANGES_REQUESTED** [review]
Hi team, 👋 

### Great job thus far! 🙌 Please see my comments. 👇 

#### Views
- [x] new => the `new.html.erb` should also contain commented out forms of how you used `<form> ` and `form_tag`.

**Kindly submit another code review request once these issues have been fixed.** _Happy coding!_ 🎉

File: app/models/user.rb (hunk 1 above)
Comment (ferbaco86) [reply app/models/user.rb]: Thanks for the suggestion. It's been fixed

File: app/controllers/users_controller.rb (hunk 2 above)
Comment (ferbaco86) [reply app/controllers/users_controller.rb]: Thanks for the suggestion. It's been fixed

File: app/controllers/users_controller.rb (hunk 3 above)
Comment (ferbaco86) [reply app/controllers/users_controller.rb]: Thanks for the suggestion. It's been fixed

File: app/views/users/edit.html.erb (hunk 4 above)
Comment (ferbaco86) [reply app/views/users/edit.html.erb]: Thanks for the suggestion. It's been fixed

File: app/controllers/users_controller.rb (hunk 5 above)
Comment (ferbaco86) [reply app/controllers/users_controller.rb]: Thanks for the suggestion. It's been fixed

File: app/controllers/users_controller.rb (hunk 6 above)
Comment (ferbaco86) [reply app/controllers/users_controller.rb]: Thanks for the suggestion. It's been fixed

File: app/views/users/edit.html.erb (hunk 7 above)
Comment (ferbaco86) [reply app/views/users/edit.html.erb]: Thanks for the suggestion. It's been fixed

File: app/views/users/new.html.erb (hunk 8 above)
Comment (ferbaco86) [reply app/views/users/new.html.erb]: Thanks for the suggestion. It's been fixed

File: app/views/users/new.html.erb [hunk 9]
```diff
@@ -0,0 +1,45 @@
+<!--<form accept-charset="UTF-8" action="/users" method="post">
```
Comment (euqueme) [inline app/views/users/new.html.erb]: - [x] uses users_path helper for action

- change action="/users" to action=users_path

**Review by euqueme — APPROVED** [review]
### Pull Request under the review by Maria Eugenia Queme P. (TSE)

### Hello @LoboArkano and @ferbaco86 :balloon: :lollipop: 

### Thank you for doing the changes requested by the previous reviewer :medal_sports: :medal_sports: I think you've met the objectives of the project, there is only one change to make in your commented forms but I won't keep you here any longer please do the change and merge this branch

You can see it in your comments below :point_down: 

### That's it, team! :raised_hands: 

### Please, keep up the good work :clinking_glasses: 

PROJECT STATUS: APPROVED :green_apple:

File: app/views/users/new.html.erb (hunk 9 above)
Comment (ferbaco86) [reply app/views/users/new.html.erb]: Change implemented! Thanks for the commentary.
Happy coding

**Pull request merged**